corehq/apps/accounting/models.py

```diff
@@ -1060,6 +1062,7 @@ def update_subscription(self, date_start=None, date_end=None,
             self.date_delay_invoicing = date_delay_invoicing
 
         self.do_not_invoice = do_not_invoice
+        self.no_invoice_reason = no_invoice_reason
```

Comment:
I'd use the same logic here as in https://github.com/dimagi/commcare-hq/pull/8396/files#diff-0cd6898a4a91ed249c85e07a44019510R1117